### ISSUE TITLE
Bug: Two windows with the same name are counted as one by getAllWindows()/getAllTitles()

### ISSUE THREAD

**Issue by macdeport** [Kalmat/PyWinCtl#3]
The title tell the story...

**Comment by Kalmat** [comment]
Even more weird... But possible... everything is possible in IT HAHAHAHAHA!! Can you please tell me which app they belong to and which title they have? There were more open windows or just those two?

Thanks!!

**Comment by macdeport** [comment]
You can reproduce this bug easily:

- Open an instance of _Path Finder_ (_PF_)
- From its windows select an arbitrary one: its name (title) in the path of this directory
- Use the dedicated _PF_ command to clone this windows: its name is of course the previous one
- Execute the "`getAllWindows()/getAllTitles()`" test

**Comment by Kalmat** [comment]
Thank you!

I am not able to reproduce it on Yosemite. I have opened several "clones", but they all are returned by getAllWindows() and getAllTitles().

I have also added several new methods: getAllApps(), getAllAppsWindows() and getAppName(). I realized that original module was focused on windows only, but it doesn't have to be the case for this new module.

Regarding this, the output (python dict) of getAllAppsWindows() is:
`{'Finder': ['Documents'], 'Safari': [], 'Path Finder': ['', 'Desktop', '', 'Applications', '', 'Applications', '', 'Documents', '', 'Documents', '', 'Downloads'], 'Terminal': ['pywinctl — osascript — 80×24'], 'Contacts': ['Contacts', '', '']}`

Several things to take into account:
- 'Applications' and 'Documents' are clones of PF windows. I also opened 'Documents' in Finder just to check.
- PF creates "fake" windows with empty ('') titles
- Contacts has also empty ('') titles, but they correspond to "real" open windows (contact cards, which seem to have no title). So, it's impossible (and not recommendable) to filter them
- Safari in this dump has no window. This is the case you were pointing out: it appears as open on the dock, but with no window associated, so it shouldn't be returned on getAllWindows() nor getAllTitles()
- If you open a window in Safari, and then minimize it (not close it), it does shows up in getAllWindows() and getAllTitles()

This is the complete dump for all methods:

GETALLWINDOWS (printing title from MacOSWindow() title property):
windows number = 18
['Documents', 'Apple', '', 'Desktop', '', 'Applications', '', 'Applications', '', 'Documents', '', 'Documents', '', 'Downloads', 'pywinctl — osascript — 80×24', 'Contacts', '', '']

GETALLTITLES:
windows number = 18
['Documents', 'Apple', '', 'Desktop', '', 'Applications', '', 'Applications', '', 'Documents', '', 'Documents', '', 'Downloads', 'pywinctl — osascript — 80×24', 'Contacts', '', '']

GETALLTITLESB (simple. straight-forward script using "repeat", used only for testing purposes since it's way slower)
windows number = 18
['Documents', 'Apple', '', 'Desktop', '', 'Applications', '', 'Applications', '', 'Documents', '', 'Documents', '', 'Downloads', 'pywinctl — osascript — 80×24', 'Contacts', '', '']

GETALLAPPS
apps number = 5
['Finder', 'Safari', 'Path Finder', 'Terminal', 'Contacts']

GETALLAPPSWINDOWS
apps number = 5
windows number = 18
{'Finder': ['Documents'], 'Safari': ['Apple'], 'Path Finder': ['', 'Desktop', '', 'Applications', '', 'Applications', '', 'Documents', '', 'Documents', '', 'Downloads'], 'Terminal': ['pywinctl — osascript — 80×24'], 'Contacts': ['Contacts', '', '']}

I'm uploading this new version (0.0.23), though it's still incomplete since I have to "propagate" all these new methods to the rest of platforms.
